apps/vmq_plugin/src/vmq_plugin_util.erl

```diff
@@ -0,0 +1,129 @@
+-module(vmq_plugin_util).
+
+-export([check_modifiers/2]).
+
+check_modifiers(auth_on_subscribe, Modifiers) ->
+    %% auth_on_subscribe take a special form
+    lists:foldl(fun (_, error) -> error;
+                    ([T, Q], Acc) when is_binary(T) and is_number(Q) ->
+                        case vmq_topic:validate_topic(subscribe, T) of
+                            {ok, Topic} ->
+                                [{Topic, to_internal_qos(round(Q))}|Acc];
```

Comment:
@dergraf the `to_internal_qos/1` function is the only change from the original `check_modifiers/2` function.